### DISCUSSION TITLE
Unexpected data_type conversion in table

### DISCUSSION THREAD

**Discussion by UK-17** [h2oai/wave#1226] [Q&A] [answered]
Was making a JIRA issue tracker using tables.
I have got a column called release which takes values like (10.0.1, 9.0.16 etc.)

The problem arises when it takes values like (10.1.5), it automatically gets rendered as datetime data type even when I have specifically defined data_type='string' for the 'release' column.

`return [
    ui.table_column(name='id',label='Issue', searchable=True,sortable=True,max_width='100'),
    ui.table_column(name='release',label='Release',filterable=True,max_width='80',data_type='string'),
    ui.table_column(name='summary',label='Summary',searchable=True,max_width='600'),
    ui.table_column(name='status',label='Status',filterable=True,max_width='100'),
    ui.table_column(name='qa_assignee',label='QA Assignee',filterable=True),
    ui.table_column(name='priority',label='Priority',filterable=True,max_width='80'),
    ui.table_column(name='components',label='Components',filterable=True,sortable=True)]`

It shows as a string when I filter the table, but as soon as I use group by release it converts.

Problem Demo : 

https://drive.google.com/file/d/17ajmcWCGWkOZ1oILYH0u_BY-luEsXCPQ/view?usp=sharing

**Comment by mturoci** [comment]
Hi @UK-17!

Can you please create a minimal repro code we can have a look at?

**Comment by UK-17** [comment]
I will put it up sure, but for meanwhile my issue was just that string like '10.1.5' are getting converted to dates on group by in the table even when i have assigned the column data_type='string'

**Comment by UK-17** [comment]
@mturoci here is a simple python script which explains my problem. 

Try grouping this table on issue.

```py
from h2o_wave import main, app, Q, ui
from random import randrange


import urllib3

urllib3.disable_warnings()

_id = 0

issues = [f'0{randrange(1,9)}.0{randrange(1,9)}.0{randrange(1,9)}' for i in range(25)]

def get_issue_rows(issues: list):
    return [ui.table_row(name=issue, cells=[issue]) for issue in issues]

@app('/testing')
async def serve(q: Q):

    columns = [ui.table_column(name='id', label='Issue', searchable=True, sortable=True, max_width='100',data_type='string')]

    q.page['issue_table'] = ui.form_card(
        box='1 7 9 10',
        items=[
            ui.table(
                name='issues',
                columns=columns,
                rows=get_issue_rows(issues),
                groupable=True,
                downloadable=True,
                resettable=True,
                height='800px')
        ]
    )
    await q.page.save()
```

**Comment by mturoci** [comment, marked as answer]
Thanks @UK-17! This is definitely a bug, tracked in https://github.com/h2oai/wave/issues/1233.
